carts/views.py

```diff
@@ -1,3 +1,30 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Sum, F
+
+from carts.models import Cart
+from users.models import User
+from products.models import Product, Size
+from users.utils import login_deco
+
+class CartView(View):
+    @login_deco
+    def get(self, request):            
+        carts = Cart.objects.filter(user_id = request.user.id)
+        total_price = carts.aggregate(price = Sum(F("count") * F("product__price")))
+
+        response = [{
+            "cart_id" : cart.id,
+            "name"    : cart.product.name,
+            "image"   : cart.product.image_url,
+            "color"   : cart.product.color.name,
+            "size"    : cart.size.name,
+            "count"   : cart.count,
+            "price"   : int(cart.count * cart.product.price)}
+            for cart in carts]
+
+        response.append({"total_price" : int(total_price["price"])})
+
+        return JsonResponse({"response":response}, status = 200)
```

Comment:
응답의 구조가 조금 부자연스럽습니다. 총액이 다시 dictionary에 들어갈 필요는 없을 것 같습니다.
아래와 같은 구조로 응답이 나오게 수정해 보세요.
```suggestion
{
    carts = [{},{},...,{}],
    total_price = 10000
}
      
```

Reply:
위 리뷰와 함께 total_price 부분 딕셔너리 형태로 추가하였습니다!

Reply:
굳~~~~~!!!